scripts/twister

```diff
@@ -685,6 +685,10 @@ structure in the main Zephyr tree: boards/<arch>/<board_name>/""")
         help="Execute time-consuming test cases that have been marked "
              "as 'slow' in testcase.yaml. Normally these are only built.")
 
+    parser.add_argument(
+        "--seed", type=int, default=SEED_INVALID,
```

Comment:
If you don't give it a default, it will be `None` when not provided, which would allow you to not need `SEED_INVALID` or `seed_valid`.
```diff
diff --git a/scripts/pylib/twister/twisterlib.py b/scripts/pylib/twister/twisterlib.py
index 8882b91284..54bf59a2e6 100755
--- a/scripts/pylib/twister/twisterlib.py
+++ b/scripts/pylib/twister/twisterlib.py
@@ -486,8 +486,7 @@ class BinaryHandler(Handler):
         self.asan = False
         self.ubsan = False
         self.coverage = False
-        self.seed_valid = False
-        self.seed = 0;
+        self.seed = None;
 
     def try_kill_process_by_pid(self):
         if self.pid_fn:
@@ -565,7 +564,7 @@ class BinaryHandler(Handler):
                        ] + command
             run_valgrind = True
 
-        if self.seed_valid:
+        if self.seed != None:
             command = command + ["--seed="+str(self.seed)];
 
         logger.debug("Spawning process: " +
@@ -2646,8 +2645,8 @@ class ProjectBuilder(FilterBuilder):
                     more_info = "build"
 
                 if ( instance.status in ["error", "failed", "timeout", "flash_error"]
-                     and hasattr(self.instance.handler, 'seed_valid')
-                     and self.instance.handler.seed_valid ):
+                     and hasattr(self.instance.handler, 'seed')
+                     and self.instance.handler.seed != None ):
                     more_info += "/seed: " + str(self.seed)
 
             logger.info("{:>{}}/{} {:<25} {:<50} {} ({})".format(
@@ -2727,11 +2726,10 @@ class ProjectBuilder(FilterBuilder):
             if instance.handler.type_str == "device":
                 instance.handler.suite = self.suite
 
-            if(self.seed != SEED_INVALID and instance.platform.name == "native_posix"):
+            if(self.seed != None and instance.platform.name == "native_posix"):
                 self.parse_generated()
                 if('CONFIG_FAKE_ENTROPY_NATIVE_POSIX' in self.defconfig and
                     self.defconfig['CONFIG_FAKE_ENTROPY_NATIVE_POSIX'] == 'y'):
-                    instance.handler.seed_valid = True
                     instance.handler.seed = self.seed
 
             instance.handler.handle()
diff --git a/scripts/twister b/scripts/twister
index ad81327d0e..8049505574 100755
--- a/scripts/twister
+++ b/scripts/twister
@@ -686,7 +686,7 @@ structure in the main Zephyr tree: boards/<arch>/<board_name>/""")
              "as 'slow' in testcase.yaml. Normally these are only built.")
 
     parser.add_argument(
-        "--seed", type=int, default=SEED_INVALID,
+        "--seed", type=int,
         help="Seed for native posix pseudo-random number generator")
 
     parser.add_argument(
```

Reply:
Thanks @aescolar! This will make it look cleaner.